Atomic/syscontainers.py

```diff
@@ -1369,6 +1370,53 @@ def uninstall(self, name):
         if rpm_installed:
             RPMHostInstall.uninstall_rpm(rpm_installed.replace(".rpm", ""))
 
+    # get the corresponding layers from an image
+    def get_layers_from_image(self,img,repo):
+        try:
+            _, rev = self._resolve_image(repo, img,allow_multiple=True)[0]
+        except (IndexError, TypeError):
+            raise ValueError("Image {} not found".format(img))
+        manifest = self._image_manifest(repo, rev)
+        if not manifest:
+            return None
+        layers = SystemContainers.get_layers_from_manifest(json.loads(manifest))
+        return layers
+
+    # define a function to handle layer image checking
+    def get_dangling_system_images(self):
```

Comment:
yes, I agree with @yuqi-zhang.  It seems that this function is doing the same thing as we already do in `prune_ostree_images`.  Could we refactor `prune_ostree_images` into two function `_check_used_layers` and `prune_ostree_images` that uses it?  Then we can use `_check_used_layers` here.

Alternatively you can refactor it directly in `get_dangling_system_images` and adapt `prune_ostree_images` to use it.